pkgs/dart_model/lib/src/type.dart

```diff
@@ -149,11 +149,11 @@ final class InterfaceType extends StaticType {
     );
   }
 
-  bool get isDartCoreObject => name == objectName;
-  bool get isDartCoreNull => name == nullName;
-  bool get isDartCoreFunction => name == functionName;
-  bool get isDartCoreRecord => name == recordName;
-  bool get isDartAsyncFutureOr => name == futureOrName;
+  bool get isDartCoreObject => name.equals(objectName);
```

Comment:
Oof yeah some way to block accidental equality checks would be great here. End users are going to see these extension types...

Reply:
Yes indeed, filed #65.